src/server/scripts/Northrend/AzjolNerub/AzjolNerub/boss_anubarak.cpp

```diff
@@ -21,351 +21,673 @@
-                    case IMPALE_PHASE_TARGET:
-                        if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 100, true))
+                        std::list<Creature*> triggers;
+                        me->GetCreatureListWithEntryInGrid(triggers, NPC_WORLD_TRIGGER);
+                        if (!triggers.empty())
                         {
-                            if (Creature* impaleTarget = DoSummonImpaleTarget(target))
-                                impaleTarget->CastSpell(impaleTarget, SPELL_IMPALE_SHAKEGROUND, true);
-                            ImpaleTimer = 3*IN_MILLISECONDS;
-                            ImpalePhase = IMPALE_PHASE_ATTACK;
+                            std::list<Creature*>::iterator it = triggers.begin();
+                            std::advance(it, urand(0, triggers.size()-1));
+                            (*it)->CastSpell(*it, SPELL_SUMMON_DARTER, true);
+                            events.Repeat(Seconds(11));
                         }
+                        else
+                            EnterEvadeMode(EVADE_REASON_OTHER);
                         break;
-                    case IMPALE_PHASE_ATTACK:
-                        if (Creature* impaleTarget = ObjectAccessor::GetCreature(*me, ImpaleTarget))
+                    }
+                    case EVENT_ASSASSIN:
+                        if (Creature* trigger = ObjectAccessor::GetCreature(*me, _assassinTrigger))
                         {
-                            impaleTarget->CastSpell(impaleTarget, SPELL_IMPALE_SPIKE, false);
-                            impaleTarget->RemoveAurasDueToSpell(SPELL_IMPALE_SHAKEGROUND);
+                            trigger->CastSpell(trigger, SPELL_SUMMON_ASSASSIN, true);
+                            trigger->CastSpell(trigger, SPELL_SUMMON_ASSASSIN, true);
+                            if (_assassinCount -= 2)
+                                events.Repeat(Seconds(20));
                         }
-                        ImpalePhase = IMPALE_PHASE_DMG;
-                        ImpaleTimer = 1*IN_MILLISECONDS;
+                        else // something went wrong
+                            EnterEvadeMode(EVADE_REASON_OTHER);
                         break;
-                    case IMPALE_PHASE_DMG:
-                        if (Creature* impaleTarget = ObjectAccessor::GetCreature(*me, ImpaleTarget))
-                            me->CastSpell(impaleTarget, SPELL_IMPALE_DMG, true);
-                        ImpalePhase = IMPALE_PHASE_TARGET;
-                        ImpaleTimer = 9*IN_MILLISECONDS;
+                    case EVENT_GUARDIAN:
+                        if (Creature* trigger = ObjectAccessor::GetCreature(*me, _guardianTrigger))
+                        {
+                            trigger->CastSpell(trigger, SPELL_SUMMON_GUARDIAN, true);
+                            trigger->CastSpell(trigger, SPELL_SUMMON_GUARDIAN, true);
+                            if (_guardianCount -= 2)
```

Comment:
please stop doing assignments inside conditional statements if they do not declare the variable in the same place as well `if (Type var = something)` is ok but `if (pre_existing_var = something)` is bad